lib/net.js

```diff
@@ -180,6 +181,9 @@ function createServer(options, connectionListener) {
 function connect(...args) {
   const normalized = normalizeArgs(args);
   const options = normalized[0];
+  // options.port === null will be checked later.
+  if (options.port === undefined && options.path == null)
+    throw new ERR_MISSING_ARGS(['options', 'port', 'path']);
```

Comment:
I believe this check should actually happen `net.socket.connect()`. Notice that the problematic behavior described in the PR is occurs when calling `net.socket.connect()` without proper arguments. Moving this piece of code there will actually fix behavior of both API fucntions, as `net.connect()` calls `net.socket.connect()` (see the return value of `net.connect()`.

Reply:
Good call, upated.